### PR TITLE
#1059 Doubled Stocktake rows

### LINKED ISSUE #1059: Double counted stocktake lines
A new uncounted line shows up for every counted line in a stocktake that has a stock line. 

Also, when first adding an item all the lines are blank and so you'd have no idea which to actually add to the stocktake. Not fully sure what to do about this!

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1059 

- When inserting a stocktake line which was counting an existing stockline, the stock line ID was not being sent, so the API assumed this was a 'new' row created for the item.
- Then I did some refactoring and shuffling things around..

### CODE REVIEW

File: packages/common/src/utils/arrays/ArrayUtils.ts [hunk 1]
```diff
@@ -34,6 +35,20 @@ export const ArrayUtils = {
   ): number => {
     return arr.reduce((sum, someEntity) => sum + someEntity[key], 0);
   },
+  immutableUpdateById: <T extends RecordWithId>(
```
Comment (mark-prins) [inline packages/common/src/utils/arrays/ArrayUtils.ts:38]: 😁

**Review by mark-prins — APPROVED** [review]
🎉  nice! 
and the hooks split out too.. an added bonus 😁

File: packages/common/src/utils/arrays/ArrayUtils.ts (hunk 1 above)
Comment (joshxg) [reply packages/common/src/utils/arrays/ArrayUtils.ts:38]: I'm unsure about the name! It's too long :(

Was thinking of some alternatives like `updateWithPatch`.. `patchUpdate` .. 🙄

Comment (mark-prins) [reply packages/common/src/utils/arrays/ArrayUtils.ts:38]: `patch` can be a verb too so you could shorten by dropping the update, if that's not confusing?
- `patchById`
- `patchImmutable`

Comment (joshxg) [reply packages/common/src/utils/arrays/ArrayUtils.ts:38]: Oh, yes! nice, thanks!

**Pull request merged**